apps/hash-graph/lib/graph/src/shared/identifier/time.rs

```diff
@@ -1,23 +1,26 @@
 mod axis;
 mod bound;
 mod interval;
-mod projection;
+mod temporal_axes;
 mod timestamp;
 
 pub use self::{
-    axis::{DecisionTime, ProjectedTime, TemporalTagged, TimeAxis, TransactionTime},
+    axis::{DecisionTime, TemporalTagged, TimeAxis, TransactionTime, VariableAxis},
     bound::{
-        IncludedTimeIntervalBound, LimitedTimeIntervalBound, TimeIntervalBound,
-        UnboundedOrExcludedTimeIntervalBound,
+        InclusiveTemporalBound, LimitedTemporalBound, TemporalBound,
+        UnboundedOrExclusiveTemporalBound,
     },
-    interval::UnresolvedTimeInterval,
-    projection::{
-        DecisionTimeImage, DecisionTimeKernel, DecisionTimeProjection, Image, Kernel,
-        TimeProjection, TransactionTimeImage, TransactionTimeKernel, TransactionTimeProjection,
-        UnresolvedDecisionTimeImage, UnresolvedDecisionTimeKernel,
-        UnresolvedDecisionTimeProjection, UnresolvedImage, UnresolvedKernel, UnresolvedProjection,
-        UnresolvedTimeProjection, UnresolvedTransactionTimeImage, UnresolvedTransactionTimeKernel,
-        UnresolvedTransactionTimeProjection,
+    interval::UnresolvedTemporalInterval,
+    temporal_axes::{
+        PinnedTemporalAxis, TemporalAxes, UnresolvedPinnedTemporalAxis, UnresolvedTemporalAxes,
+        UnresolvedVariableTemporalAxis, VariableTemporalAxis,
     },
     timestamp::Timestamp,
 };
+use crate::interval::Interval;
+
+pub type TemporalInterval<A> = Interval<Timestamp<A>, TemporalBound<A>, TemporalBound<A>>;
+pub type LimitedTemporalInterval<A> =
+    Interval<Timestamp<A>, TemporalBound<A>, LimitedTemporalBound<A>>;
+pub type VersionInterval<A> =
```

Comment:
This should probably have a doc-comment. 

I'm also not sure about the naming of this one, is there a reason we aren't just staying consistent with names with the BP implementation? It reduces cognitive burden and means we don't have to spend time discussing these things multiple times.

Reply:
As spoken offline I renamed `LimitedTemporalInterval` to `RightBoundedTemporalInterval` and `VersionInterval` to `LeftClosedTemporalInterval`: [Rename intervals to better match their bounds](https://github.com/hashintel/hash/pull/1854/commits/572155915e0370e32ac4a5edaa25f850fa94f598)